src/transport/mod.rs

```diff
@@ -19,11 +20,12 @@ pub use self::{
 
 fn set_nodelay_or_warn(socket: &::tokio::net::TcpStream) {
     if let Err(e) = socket.set_nodelay(true) {
-        tracing::warn!(
-            "could not set TCP_NODELAY on {:?}/{:?}: {}",
-            socket.local_addr(),
-            socket.peer_addr(),
-            e
-        );
+        tracing::warn!("failed to set nodelay: {}", e);
+    }
+}
+
+pub fn set_keepalive_or_warn(tcp: &mut TcpStream, ka: Option<Duration>) {
```

Comment:
Nit: AFAICT this [doesn't require a mut reference to the `TcpStream`](https://docs.rs/tokio/0.1.12/tokio/net/struct.TcpStream.html#method.set_keepalive)?
```suggestion
pub fn set_keepalive_or_warn(tcp: &TcpStream, ka: Option<Duration>) {
```